dsgrid/registry/dataset_registry_manager.py

```diff
@@ -127,37 +143,123 @@ def _check_lookup_data_consistency(self, config: DatasetConfig, load_data_lookup
+        if len(dimension_types) != len(expected_dimensions):
+            raise DSGInvalidDataset(
+                f"load_data_one_table does not have the correct number of dimensions specified between trivial and non-trivial dimensions."
             )
+
+        missing_dimensions = set(expected_dimensions).difference(dimension_types)
+        if len(missing_dimensions) != 0:
             raise DSGInvalidDataset(
-                f"load_data timestamps do not match expected times. mismatch={mismatch}"
+                f"load_data_one_table is missing dimensions: {missing_dimensions}. If these are trivial dimensions, make sure to specify them in the Dataset Config."
             )
 
-        timestamps_by_id = (
-            load_data.select("timestamp", "id")
-            .groupby("id")
-            .agg(F.countDistinct("timestamp").alias("distinct_timestamps"))
-        )
-        for row in timestamps_by_id.collect():
-            if row.distinct_timestamps != len(expected_timestamps):
+        dimension_types = [
+            d for d in dimension_types if d != DimensionType.TIME
+        ]  # remove time type
+        for dimension_type in dimension_types:
+            name = dimension_type.value
+            dimension = config.get_dimension(dimension_type)
+            dim_records = dimension.get_unique_ids()
+            if dimension_type == DimensionType.METRIC:
+                data_records = set(metric_cols)
+            else:
+                data_records = get_unique_values(load_data_one_table, name)
+            if dim_records != data_records:
+                logger.error(
+                    "Mismatch in load_data_one_table records. dimension=%s mismatched=%s",
+                    name,
+                    data_records.symmetric_difference(dim_records),
+                )
+                raise DSGInvalidDataset(
+                    f"load_data_one_table records do not match dimension records for {name}"
+                )
+
+    def _check_dataset_time_consistency(self, config: DatasetConfig, load_data):
+        """
+        Common time check.
+        Only check time if it's not ANNUAL time type
+        """
+        time_dim = config.get_dimension(DimensionType.TIME)
+
+        if time_dim.model.time_type != TimeDimensionType.ANNUAL:
+            tz = time_dim.get_tzinfo()
+            time_ranges = time_dim.get_time_ranges()
+            assert len(time_ranges) == 1, len(time_ranges)
+            time_range = time_ranges[0]
+            # TODO: need to support validation of multiple time ranges: DSGRID-173
+
+            expected_timestamps = time_dim.list_time_range(time_range)
+            actual_timestamps = [
+                x.timestamp
+                for x in load_data.select("timestamp").distinct().sort("timestamp").collect()
+            ]
+            try:
+                actual_timestamps = [x.astimezone().astimezone(tz) for x in actual_timestamps]
+            except AttributeError:
```

Comment:
Why will this be raised? Can that condition be checked explicitly?

Reply:
Yes, in the case of annual time, x is an integer, not a date time obj, so it will fail at x.astimezone() with AttributeError.

Reply:
It would be much safer to handle it explicitly. Relying on exceptions can result in very strange behaviors if something happens that you don't expect - like if AttributeError is raised for a different reason.